src/Controllers/Ops.php

```diff
@@ -101,40 +102,21 @@ protected function saveToFile($file, $data, $package_name = null)
         return true;
     }
 
-    protected function recSave($pre_format, $key, $code, $v)
+    private static function explodeTranlsationKey(string $nestedTranslationKey, string|null $value): array
     {
-        if (strpos($key, '.')) {
-            $assoc = [];
-            $exp   = explode('.', $key);
+        $doneSplitting = !Str::contains($nestedTranslationKey, '.');
 
-            // convert array dot to associative
-            while (!empty($exp)) {
-                $assoc = [array_pop($exp) => $assoc];
-            }
-
-            // set value to last key
-            Arr::set($assoc, $key, $v);
-
-            // get root key
-            $first_key = strtok($key, '.');
-
-            // hacky fix to avoid indexed nesting
-            $final = array_pop($assoc);
-            $a     = key($final);
-            $b     = current($final);
+        if ($doneSplitting) return [$nestedTranslationKey => $value];
 
-            // 3rd level deep
-            if (is_array($b)) {
-                $c = key($b);
-                $d = current($b);
+        $currentLevel = collect(explode('#', Str::replaceFirst('.', '#', $nestedTranslationKey)));
+        $collection = collect([$currentLevel[0] => self::explodeTranlsationKey($currentLevel[1], $value)]);
+        return $collection->toArray();
+    }
 
-                $pre_format[$code][$first_key][$a][$c] = $d;
-            } else {
-                $pre_format[$code][$first_key][$a] = $b;
-            }
-        } else {
-            $pre_format[$code][$key] = $v;
-        }
+    protected function recSave(array $pre_format, string $key, string $code, string|null $v)
+    {
+        $codeContent = collect($pre_format)->get($code, []);
+        $pre_format[$code] = collect($codeContent)->mergeRecursive(self::explodeTranlsationKey($key, $v))->toArray();
```

Comment:
Set the language to a new array that is recursively merged from the existing array and the newly exploded key.